packages/react-components/src/InputNumber.tsx

```diff
@@ -168,12 +168,18 @@ function getValuesFromString (value: string, si: SiDef | null, props: Props): [s
   // Sometimes the value is already formatted, avoid formatting in those cases
   const regex = RegExp('^\\d{1,3}(,\\d{3})*(\\.\\d+)?$');
   // Format only if required
-  if (!regex.exec(value)) {
-    const defaultValue = valueBn
-      ? formatBalance(valueBn, { forceUnit: '-', withSi: false }).replace(',', ',')
-      : valueBn;
-    formatedValue = defaultValue.split('.')[0];
+  if (value && !regex.exec(value)) {
+    let decimalValue;
+    if (value.includes('.')) {
+      decimalValue = value.split('.');
+      value = decimalValue[0];
+    }
+    formatedValue = formatNumber(new BN(value.replace(/,/g, '')));
+    formatedValue = decimalValue ? `${formatedValue}.${decimalValue[1]}` : formatedValue;
   }
+  // The following will just remove the 0 on the left, example 01 -> 1
+  formatedValue = value.length === 2 ? new BN(value).toString() : formatedValue;
```

Comment:
We may need to consider edge situations like `010`, `010.`, `010.0`, `010.1`;
What if using `value.startWith('0')` to do the condition check?

Reply:
May worth exporting the whole function, and add unit tests to cover possible edge conditions?